PusherServer.Tests/AcceptanceTests/Trigger.cs

```diff
@@ -175,6 +176,71 @@ public void It_should_return_a_200_response_async()
     }
 
     [TestFixture]
+    public class When_Triggering_a_Batch_of_Events
+    {
+        [Test]
+        public void It_should_return_a_200_response()
+        {
+            IPusher pusher = new Pusher(Config.AppId, Config.AppKey, Config.AppSecret, new PusherOptions()
+            {
+                HostName = Config.Host
+            });
+
+            BatchEvent[] events = new BatchEvent[] {
+                new BatchEvent {
+                    channel = "my-channel-1",
+                    name = "my_event",
+                    data = new JsonSerializer().Serialize(new { hello = "world" })

```

Comment:
I don't like that I have to push serialisation back to the user here - what's the correct way to deal with this slickly. Ideally, we want to supply a dictionary when constructing the BatchEvent, but serialise it to a string and have the type of the property be `string` (for the later serialisation pass).

Can this be done together with the slick property initialisation syntax, or do I need to expose different objects to the end user and then translate in to `BatchEvent`s for serialisation?